src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -17,6 +17,7 @@ internal class WinHttpResponseStream : Stream
         private volatile bool _disposed;
         private readonly WinHttpRequestState _state;
         private SafeWinHttpHandle _requestHandle;
+        private CancellationTokenRegistration _ctr;
```

Comment:
Registering with a token stores the delegate and object state into the token, so that the token can invoke the callback when cancellation occurs. As such, as long as the token it's rooted, this state will be as well until either cancellation is requested or the registration is disposed. If you know that's not going to be problem, e.g. because any possible token won't be rooted arbitrarily long, then disposing isn't necessary, but generally such guarantees aren't known. You'd know best if this is the special care where it's not necessary.